### PR TITLE
feat: add prisma support

### PR DESCRIPTION
**Pull request by elianiva**
Still need to wait for https://github.com/LumaKernel/tree-sitter-prisma/pull/2, in the meantime we can use my fork to make it easier for those who want to try this out.

### CODE REVIEW

**Comment by theHamsta** [comment]
You could add a `requires_generate_from_source = true`

**Comment by theHamsta** [comment]
@elianiva can you rebase? Are you willing to maintain your fork when we leave it the way it is right now?

**Comment by elianiva** [comment]
Yes, there's not a lot of work left for the parser, it's just a matter of nvim-treesitter compatiblity.

**Comment by theHamsta** [comment]
@elianiva 
```
```
nvim-treesitter[prisma]: Error during `npm install` (required for parser generation of prisma with npm dependencies)
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR! 
npm ERR! While resolving: tree-sitter-prisma@0.1.0
npm ERR! Found: eslint@7.32.0
npm ERR! node_modules/eslint
npm ERR!   dev eslint@"^7.16.0" from the root project
npm ERR!   peer eslint@"^5.0.0 || ^6.0.0 || ^7.0.0" from @typescript-eslint/eslint-plugin@4.33.0
npm ERR!   node_modules/@typescript-eslint/eslint-plugin
npm ERR!     dev @typescript-eslint/eslint-plugin@"^4.10.0" from the root project
npm ERR!     peer @typescript-eslint/eslint-plugin@"^4.6.1" from eslint-config-typestrict@1.0.2
npm ERR!     node_modules/eslint-config-typestrict
npm ERR!       dev eslint-config-typestrict@"^1.0.1" from the root project
npm ERR!   1 more (@typescript-eslint/parser)
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer eslint@"^3.0.0 || ^4.0.0 || ^5.0.0 || ^6.0.0" from eslint-plugin-sonarjs@0.5.0
npm ERR! node_modules/eslint-plugin-sonarjs
npm ERR!   dev eslint-plugin-sonarjs@"^0.5.0" from the root project
npm ERR!   peer eslint-plugin-sonarjs@"^0.5.0" from eslint-config-typestrict@1.0.2
npm ERR!   node_modules/eslint-config-typestrict
npm ERR!     dev eslint-config-typestrict@"^1.0.1" from the root project
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR! 
npm ERR! See /home/stephan/.npm/eresolve-report.txt for a full report.
npm ERR! A complete log of this run can be found in:
npm ERR!     /home/stephan/.npm/_logs/2021-11-21T09_22_16_917Z-debug.log
```
are you also getting this error with upstream tree-sitter-prisma (same with yarn)?

How does the upstream compare to https://github.com/victorhqc/tree-sitter-prisma?

**Comment by elianiva** [comment]
yes, I also got the error. Could be fixed using `--force` but not sure if it's ideal or not.

I actually implemented from `victorhqc/tree-sitter-prisma` before but it's not as complete as Luma's one. I switched because there hasn't been much activity on that repo. Seems like the author updated the repo 9 days ago. We could probably switch to that instead?

**Comment by theHamsta** [comment]
I know that I could have used `--forced` but we shouldn't do that as a default for installation. When Luma is more complete we should use it instead. However, we want a maintained repo (e.g. issues like the `--forced` get fixed). Did you get in contact with the original or does the repo appear unmaintained. When you are willing to maintain your fork, we can of course just use your fork.

**Comment by elianiva** [comment]
I'll try to switch to victorhqc's repo later tonight, seems like it receives some updates recently. (and someone also made a query for it)

**Comment by elianiva** [comment]
@richin13 do you want to make the PR instead? since you already made the queries there and I don't feel good using someone's code on my PR. If yes, I'll close this and we'll use yours instead.

**Comment by richin13** [comment]
@elianiva I don't mind at all if you want to just update this PR :)

File: queries/prisma/highlights.scm [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+(identifier) @variable
+(string) @string
+
+"(" @punctuation.bracket
+")" @punctuation.bracket
+"[" @punctuation.bracket
+"]" @punctuation.bracket
+"{" @punctuation.bracket
+"}" @punctuation.bracket
+"=" @operator
```
Comment (theHamsta) [inline queries/prisma/highlights.scm:25]: Could you use `[ ...... ] @punctuation.bracket` here to summarize the different brackets in one alteration

**Comment by theHamsta** [comment]
@elianiva is this ready to merge?

You can also add highlight tests if you're already using example files by dropping them here https://github.com/nvim-treesitter/nvim-treesitter/tree/master/tests/query/highlights with assertions like here https://github.com/nvim-treesitter/nvim-treesitter/blob/master/tests/query/highlights/ecma/test.ts Just if you are already using example test files but this should be already ready to go.

**Comment by elianiva** [comment]
I'm pretty much done. Still not sure about the failing test, though.

**Comment by theHamsta** [comment]
The test is probably failing because there is not ftdetect for prisma. You could remove test or add a simple ftdetect or add that ftdetect to `scripts/minimal_init.lua`

**Comment by elianiva** [comment]
hmm, this node is `keyword` but the test is complaining it's an `attribute`.

image

**Comment by theHamsta** [comment]
It's because you set the filetype as `ql` which will parse the file as graphql

**Comment by theHamsta** [comment]
I'll fix the tests.

**Review by theHamsta — APPROVED** [review]

**Comment by theHamsta** [comment]
@elianiva is this PR ready to merge now?

**Comment by stsewd** [comment]
> @elianiva is this PR ready to merge now? @stsewd would ask you to add `(comment) @comment` in injections.scm

a little late, but yes :D

**Comment by pencilcheck** [comment]
Works, but you need to manually type `:TSEnable highlight` in a prisma schema file to work, not sure why...

**Comment by MuntasirSZN** [comment]
> Works, but you need to manually type `:TSEnable highlight` in a prisma schema file to work, not sure why...

Yeah true. Maybe Add A Autocmd?

**Pull request merged**